src/Lecture1.hs

```diff
@@ -94,7 +94,12 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString start end str 
+    |  start < 0 =  subStr 0 end str
+    |  end < 0 = ""
+    |  otherwise = subStr start end str
+        where subStr start end str = reverse (drop (length str - end - 1) (reverse (drop start str)))
```

Comment:
Both `length` and `reverse` functions are slow on lists 🐌 
You can use a standard function `take` to avoid reversing and dropping from the end